test/media-segment-request.test.js

```diff
@@ -766,6 +767,135 @@ QUnit.test('non-TS segment will get parsed for captions on next segment request
   this.standardXHRResponse(initReq, mp4VideoInit());
 });
 
+QUnit.test('unsupported segments will not make it to the transmuxer', function(assert) {
+  const done = assert.async();
+
+  this.transmuxer = this.createTransmuxer(true);
+
+  const calls = {};
+  const options = {
+    xhr: this.xhr,
+    xhrOptions: this.xhrOptions,
+    decryptionWorker: this.mockDecrypter,
+    segment: {
+      transmuxer: this.transmuxer,
+      resolvedUri: 'foo.ac3'
+    },
+    doneFn: () => {
+      assert.deepEqual(calls, {
+        data: 0,
+        trackInfo: 1,
+        progress: 1,
+        timingInfo: 0,
+        captions: 0,
+        id3: 0
+      }, 'only called trackinfo and done');
+      done();
+    },
+    handlePartialData: false
+  };
+
+  ['progress', 'trackInfo', 'timingInfo', 'id3', 'captions', 'data'].forEach(function(key) {
+    calls[key] = 0;
+    options[`${key}Fn`] = () => calls[key]++;
+  });
+
+  mediaSegmentRequest(options);
+
+  assert.equal(this.requests.length, 1, 'there is one request');
+
+  const segmentReq = this.requests.shift();
+
+  assert.equal(segmentReq.uri, 'foo.ac3', 'segment-request');
+  this.standardXHRResponse(segmentReq, new Uint8Array([0x00, 0x00, 0x00]));
```

Comment:
Might be good for us to use an actual ac3 file here, as it should catch the case we saw with the ID3 signature (if it ever returns).